File: CPP/BinaryToDecimal.cpp
#include <iostream>
#include <cmath>
using namespace std;
/*
	101 - 5
	111 - 7
	1111111111111111 - 65535
*/
int main(){
	unsigned long int n, res = 0;
	cin >> n;
	int p = 0;
	while(n){
		int rem = n % 10;
		res += rem*pow(2, p);
		n /=10;
		p++;
	}
	cout << res << endl;
}

File: Java/BinaryToDecimal.java
import java.util.Scanner;
/*
	101 - 5
	111 - 7
	1111111111111111 - 65535
*/
class BinaryToDecimal{
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String s = sc.next();
		long n = Long.parseUnsignedLong(s);
		int res = 0, p = 0;
		while(n!=0){
			short rem = (short)(n % 10);
			res += rem*Math.pow(2, p++);
			n /= 10;
		}
		System.out.println(res);
	}
}